src/hal/src/queue/mod.rs

```diff
@@ -42,6 +42,16 @@ pub enum QueueType {
     Transfer,
 }
 
+/// Submission information for a command queue.
+pub struct Submission<'a, B: Backend + 'a, I> {
+    /// Command buffers to submit.
+    pub command_buffers: I,
+    /// Semaphores to wait being signalled before submission.
+    pub wait_semaphores: &'a [(&'a B::Semaphore, pso::PipelineStage)],
```

Comment:
Why this is not an iterator type? This seems inconsistent with rest of API.

`WI: IntoIterator<Item = (&'a W, pso::PipelineStage)>, W: Borrow<B::Semaphore>` on methods should do :)

Reply:
holy cow, that would make `submit` to carry 7 generic parameters :/ How about doing this as a follow up? I don't think my PR regresses this anyhow.

Reply:
Lucky number :)

Reply:
Old `Submission` uses `IntoIterator` for semaphores. So this PR regress it.
But it was done suboptimal `Borrow<(&B::Semaphore, PipelineStage)>` is not very convenient.